test_utils.go

```diff
@@ -2,17 +2,46 @@ package redis
 
 import (
 	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
 	"fmt"
-
 	"github.com/mediocregopher/radix/v4"
 )
 
-func createUser(hostname string, port int, adminuser, adminpassword, username, password, aclRule string) (err error) {
-	poolConfig := radix.PoolConfig{
-		Dialer: radix.Dialer{
-			AuthUser: adminuser,
-			AuthPass: adminpassword,
-		},
+func createUser(hostname string, port int, redisTls bool, caCrt, adminuser, adminpassword, username, password, aclRule string) (err error) {
```

Comment:
Nit: Unrelated to your PR (sorry 🙂), I'd be in favor of moving this function to the `redis_test.go` file and deleting the `test_utils.go` file. Not sure if there is value in an additional file for a single function that's only used in `redis_test.go`.

Reply:
Good suggestion. I agree!